cli/server/server.go

```diff
@@ -43,36 +89,160 @@ func newGraceContext() context.Context {
 	return ctx
 }
 
-func startServer(ctx *cli.Context) error {
-	net := config.ModePrivNet
+// getConfigFromContext looks at path and mode flags in the given config and
+// returns appropriate config.
+func getConfigFromContext(ctx *cli.Context) (config.Config, error) {
+	var net = config.ModePrivNet
 	if ctx.Bool("testnet") {
 		net = config.ModeTestNet
 	}
 	if ctx.Bool("mainnet") {
 		net = config.ModeMainNet
 	}
-
-	grace, cancel := context.WithCancel(newGraceContext())
-	defer cancel()
-
 	configPath := "./config"
 	if argCp := ctx.String("config-path"); argCp != "" {
 		configPath = argCp
 	}
-	cfg, err := config.Load(configPath, net)
+	return config.Load(configPath, net)
+}
+
+// handleLoggingParams enables debugging output is that's requested by the user.
+func handleLoggingParams(ctx *cli.Context) {
+	if ctx.Bool("debug") {
+		log.SetLevel(log.DebugLevel)
+	}
+}
+
+func getCountAndSkipFromContext(ctx *cli.Context) (uint32, uint32) {
+	count := uint32(ctx.Uint("count"))
+	skip := uint32(ctx.Uint("skip"))
+	return count, skip
+}
+
+func dumpDB(ctx *cli.Context) error {
+	cfg, err := getConfigFromContext(ctx)
 	if err != nil {
 		return cli.NewExitError(err, 1)
 	}
+	handleLoggingParams(ctx)
+	count, skip := getCountAndSkipFromContext(ctx)
 
-	serverConfig := network.NewServerConfig(cfg)
+	var outStream = os.Stdout
+	if out := ctx.String("out"); out != "" {
+		outStream, err = os.Create(out)
+		if err != nil {
+			return cli.NewExitError(err, 1)
+		}
+	}
+	defer outStream.Close()
+	writer := io.NewBinWriterFromIO(outStream)
+
+	grace, cancel := context.WithCancel(newGraceContext())
+	defer cancel()
 
 	chain, err := initBlockChain(cfg)
+	if err != nil {
+		return cli.NewExitError(err, 1)
+	}
+	go chain.Run(grace)
+
+	cheight := chain.BlockHeight()
```

Comment:
cheight? =) only after a few seconds realized that it's chainHeight

Reply:
I'll rename it.